frontend/src/pages/SignUp/PasswordInput.tsx

```diff
@@ -0,0 +1,55 @@
+import { HTMLAttributes, useState } from 'react';
+import InputField from '@components/common/InputField';
+import { FaEyeSlash, FaEye } from 'react-icons/fa';
+import S from './style';
+
+export default function PasswordInput(props: HTMLAttributes<HTMLInputElement>) {
+  const [isShow, setIsShow] = useState(false);
+  const [isCapsLockOn, setIsCapsLockOn] = useState(false);
+  const [isNumLockOn, setIsNumLockOn] = useState(false);
+
+  const handleIconClick = () => {
+    setIsShow((prev) => !prev);
+  };
+
+  const setModifierState = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    setIsCapsLockOn(event.getModifierState('CapsLock'));
+    setIsNumLockOn(event.getModifierState('NumLock'));
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    setModifierState(event);
+    // eslint-disable-next-line react/destructuring-assignment
+    if (props?.onKeyUp) props.onKeyUp(event);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    setModifierState(event);
+    // eslint-disable-next-line react/destructuring-assignment
+    if (props?.onKeyDown) props.onKeyDown(event);
+  };
+
+  const handelBlur = (event: React.FocusEvent<HTMLInputElement, Element>) => {
+    setIsCapsLockOn(false);
+    setIsNumLockOn(false);
+
+    // eslint-disable-next-line react/destructuring-assignment
+    if (props?.onBlur) props.onBlur(event);
+  };
+
+  return (
+    <S.PasswordInputContainer>
+      <InputField
+        label="비밀번호"
+        placeholder="비밀번호"
+        type={isShow ? 'text' : 'password'}
+        error={isCapsLockOn ? 'Caps Lock이 켜져 있습니다.' : isNumLockOn ? 'Num Lock이 켜져 있습니다.' : undefined}
```

Comment:
capsLock과 numLock까지 꼼꼼하게 처리해주셨네요👍
다만 숫자 키패드의 경우 사용자가 키패드로 입력을 시도하는대 안될 때 피드백을 줘야 하니 아래와 같이 변경해보는 건 어떨까요?

```suggestion
        error={isCapsLockOn ? 'Caps Lock이 켜져 있습니다.' : !isNumLockOn ? '숫자 키패드가 비활성화되었습니다.' : undefined}
```